specs/lazy-adr/adr-010-exec-api.md

```diff
@@ -3,17 +3,33 @@
 ## Changelog
 
 - 2025.01.13: Initial draft
+- 2025.04.24: Various improvements
 
 ## Context
 
 Introduction of the Execution API makes rollkit very generic and execution-environment agnostic.
 It removes all ABCI-centric code for full interoperability with other types of VMs.
 
+The Execution API serves as a bridge between Rollkit and various execution environments (VMs),
+allowing Rollkit to remain agnostic to the specific implementation details of the execution layer.
+This separation enables:
+
+1. Support for multiple VM types (EVM, WASM, etc.)
+2. Easier integration with different execution environments
+3. Cleaner separation of concerns between consensus and execution
+4. More flexible and maintainable architecture
+
 ## Alternative Approaches
 
 1. Maintain current state: keep ABCI interface and implement other VMs inside ABCI application.
+   - Pros: No changes required to existing code
+   - Cons: ABCI-specific code remains, limiting flexibility
 2. Migrate to Engine API.
+   - Pros: Standard interface for EVM-based chains
+   - Cons: Too specific to EVM, not suitable for other VMs
 3. Create new API generic enough to handle any arbitrary VM.
+   - Pros: Maximum flexibility, clean separation of concerns
+   - Cons: Requires significant refactoring
```

Comment:
This is what execution interface already is. Can remove it as alternative approach.
```suggestion
```